File: ch31/App_Code/CS/Listing31_3.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Threading.Tasks;
using System.Web;


public class Listing31_3
{
	public Listing31_3()
	{
		
	}
    public async Task<int> RSSFeedLengthAsync(IList<Uri> uris)
    {
        int total = 0;
        foreach (var uri in uris)
        {            
            var data = await new WebClient().DownloadDataTaskAsync(uri);
            total += data.Length;
        }
        
        return total;
    }


}

File: ch31/App_Code/VB/Listing31_3.vb
Imports Microsoft.VisualBasic
Imports System.Threading.Tasks
Imports System.Net

Public Class Listing31_3
    Public Async Function RSSFeedLengthAsync(uris As IList(Of Uri)) As Task(Of Integer)
        Dim total As Integer = 0
        For Each uri In uris
            Dim data = Await New WebClient().DownloadDataTaskAsync(uri)
            total += data.Length
        Next
        Return total

    End Function
End Class
